packages/web-runtime/src/components/ThemeSwitcher.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+  <div>
+    <oc-select
+      :model-value="currentThemeOrAuto"
+      :clearable="false"
+      :options="availableThemesAndAuto"
+      option-label="name"
+      @update:model-value="updateTheme"
+    />
+  </div>
+</template>
+<script lang="ts">
+import { computed, defineComponent, ref, unref } from 'vue'
+import { useMessages, useThemeStore, WebThemeType } from '@ownclouders/web-pkg'
+import { storeToRefs } from 'pinia'
+import { useGettext } from 'vue3-gettext'
+
+export default defineComponent({
+  setup() {
+    const themeStore = useThemeStore()
+    const { showMessage } = useMessages()
+    const { $gettext } = useGettext()
+    const autoTheme = { name: $gettext('Auto (same as system)') }
+    const { availableThemes, currentTheme } = storeToRefs(themeStore)
+    const currentThemeSelection = ref(null)
+
+    const { setAndApplyTheme, setAutoSystemTheme, isCurrentThemeAutoSystem } = themeStore
+
+    const updateTheme = (newTheme: WebThemeType) => {
+      currentThemeSelection.value = newTheme
+      if (newTheme == autoTheme) {
+        setAutoSystemTheme()
+        return
+      }
+      setAndApplyTheme(newTheme)
+      showMessage({ title: $gettext('Preference saved.') })
+    }
+
+    const currentThemeOrAuto = computed(() => {
+      if (unref(currentThemeSelection)) {
+        return unref(currentThemeSelection)
+      }
+      if (unref(isCurrentThemeAutoSystem)) {
+        return autoTheme
+      }
+      return unref(currentTheme)
+    })
+    const availableThemesAndAuto = [autoTheme, ...unref(availableThemes)]
```

Comment:
Here as well for reactivity:

```suggestion
    const availableThemesAndAuto = computed(() => [autoTheme, ...unref(availableThemes)])
```